src/main.rs

```diff
@@ -315,19 +318,37 @@ impl Logger for LogPrinter {
 	}
 }
 
+fn create_global_args() -> ArgMan {
+	let mut g_args = ArgMan::new();
+	// Common arguments:
+	g_args.add_arg("-chain", "regtest".to_string(), "Selected chain/network to operate with");
+	g_args.add_arg_unset("-datadir", "Storage directory path");
+	g_args.add_arg("-p2pport", "9735".to_string(), "Port to listen to as a p2p lightning node");
+
+	// Per chain arguments:
+	let default_empty: HashMap<String, String> = HashMap::new();
+	g_args.add_arg_with_category("-rpcauth", default_empty, "bitcoind RPC auth (example: regtest.-rpcauth=user:pass)");
+	let mut default_host: HashMap<String, String> = HashMap::new();
+	default_host.insert("main".to_string(), "localhost:8332".to_string());
+	default_host.insert("test".to_string(), "localhost:18332".to_string());
+	default_host.insert("regtest".to_string(), "localhost:18443".to_string());
+	g_args.add_arg_with_category("-rpchost", default_host, "bitcoind RPC host to connect to");
+
+	g_args
+}
+
 fn main() {
-	println!("USAGE: rust-lightning-jsonrpc user:pass@rpc_host:port storage_directory_path [port]");
-	if env::args().len() < 3 { return; }
-
-	let rpc_client = {
-		let path = env::args().skip(1).next().unwrap();
-		let path_parts: Vec<&str> = path.split('@').collect();
-		if path_parts.len() != 2 {
-			println!("Bad RPC URL provided");
-			return;
-		}
-		Arc::new(RPCClient::new(path_parts[0], path_parts[1]))
-	};
+	let mut g_args = create_global_args();
+	if !g_args.parse_args() {
+		println!("OLD USAGE: rust-lightning-jsonrpc user:pass@rpc_host:port storage_directory_path [port]");
+		println!("NEW USAGE: rust-lightning-jsonrpc -chain=regtest -rpcauth=user:pass -datadir=storage_directory_path [-p2pport=port] [-rpc_host=rpc_host:port] ");
```

Comment:
No need to keep the old usage around, just remove that cruft!

Reply:
Done, kept an example usage, even though it is kind of redundant since any failure parsing args will produce the equivalent of ```rust-lightning-bitcoinrpc --help``` after the first parsing error.